hal/src/rtl872x/usart_hal.cpp

```diff
@@ -585,16 +644,21 @@ class Usart {
             // Disable Rx interrupt
             UART_INTConfig(uartInstance, (RUART_IER_ERBI | RUART_IER_ELSI | RUART_IER_ETOI), DISABLE);
```

Comment:
Might want to add `RX_THRESHOLD` check similar to Gen 3: https://github.com/particle-iot/device-os/blob/develop/hal/src/nRF52840/usart_hal.cpp#L557

to avoid getting into weird ring buffer configurations with wraparound.

Reply:
Added to keep consistent between Gen 3 and RTL872x.